platform-api/src/services/discord_client.py

```
@@ -0,0 +1,39 @@
+        )
+
+        # Handle upstream errors
+        if resp.status_code == 429:
+            # Forward rate limit with Retry-After header if available
+            retry_after = resp.headers.get("Retry-After", "60")
+            raise HTTPException(
+                status_code=503,
+                detail=f"Discord rate limited, retry after {retry_after}s",
+                headers={"Retry-After": retry_after}
+            )
+        elif resp.status_code >= 300:
```

Ruff (0.15.11)

[warning] Unnecessary `elif` after `raise` statement

Remove unnecessary `elif`

(RET506)